src/engine/controls/bpmcontrol.cpp

```diff
@@ -264,6 +334,22 @@ void BpmControl::slotTapFilter(double averageLength, int numSamples) {
     pTrack->trySetBeats(*newBeats);
 }
 
+void BpmControl::slotScaleBpm(mixxx::Beats::BpmScale bpmScale) {
+    const TrackPointer pTrack = getEngineBuffer()->getLoadedTrack();
+    if (!pTrack) {
+        return;
+    }
+    const mixxx::BeatsPointer pBeats = pTrack->getBeats();
+    if (!pBeats) {
+        return;
+    }
+    const auto newBeats = pBeats->tryScale(bpmScale);
+    if (!newBeats) {
+        return;
+    }
+    pTrack->trySetBeats(*newBeats);
```

Comment:
```suggestion
    pTrack->trySetBeats(newBeats);
```
`newBeats` as a pointer already.